Remove internal functions of named colors from the public API

dalexeev opened the pull request:
The following static methods are internal and became available in the public API by accident\* (as I understand it):

* `find_named_color`;
* `get_named_color_count`;
* `get_named_color_name`;
* `get_named_color`.

I suggest to hide them and add the `get_named_colors` method or other (if needed).

---

\* - I think so, because IMO these are internal methods, I do not see cases of using these methods in user code, except for:

1. getting a random named color (`Color.get_named_color(randi() % Color.get_named_color_count())`), but for these purposes we can add a separate method;
2. getting a list of named colors, but for these purposes we can add a separate method or add the ability to get a list/values of constants via `ClassDB` universally, for any class, not just for `Color`.

`find_named_color` doesn't seem to be useful at all, since the id of a named color can only be used to get its name or color. Both options are useless:

```gdscript
get_named_color_name(find_named_color(name)) # ≈ name
get_named_color(find_named_color(name)) # Color(name) or Color.from_string(name, default)
```

dalexeev committed:
Remove internal functions of named colors from the public API

Mickeon commented:
As @mhilbrunner brought in [RocketChat](https://chat.godotengine.org/channel/general?msg=rMKKnevwazNsxChH6):

> #64830, what makes you think the named colors APIs were exposed by accident?

This is probably a piece of information that got lost as months have passed.

In my personal opinion, these methods are essentially useless to the average user, and overall they clutter the documentation. If necessary, it would be better to expose the named colors by returning a Dictionary of key/values from a single function instead of resorting on these four methods.

dalexeev commented:
> these methods are essentially useless to the average user. They return integers, indexes, and essentially clutter the documentation

> fetching Colors by name can be so much simpler than all that (see `Color.from_string()`)

Yes, I agree with this. I've added a detailed explanation to OP.

> If there's need to, for example, count how many named colors there are, it would be better to expose these as a Dictionary of key/values returned from a single function

In the original version of this PR, I did it in a separate commit, but after recent changes, it's not building, so I removed it.

mhilbrunner commented:
> In my personal opinion, these methods are essentially useless to the average user. They return integers, indexes, and essentially clutter the documentation. They should simply not be encouraged, at all, because fetching Colors by name can be so much simpler than all that (see Color.from_string()).

I think the use case is iterating the list of named colors from e.g. a Editor extension or tool?
I agree exposing them as a dictionary makes more sense (and the same goes for a lot of similar places in the current API, where it has these kinds of "access/list by integer ID" APIs).

Personally, I'd be fine with the change to a dictionary, I'm also fine to remove these if we judge the use case too rare to bother, or if we do have evidence these got exposed by accident. As is, its a bit hard to judge for me.

Mickeon commented:
I'd gladly contribute to the Dictionary idea before that time, to go hand-to-hand with this PR. Even if it's a "compatibility break" between versions, it's so unlikely users have even made use of these functions just yet, especially for the reasons stated above.

dalexeev commented:
> or if we do have evidence these got exposed by accident

I have no strong evidence that these methods were exposed by accident. Just a guess. These methods were exposed in #46378, which introduced the ability to bind static methods of `Variant` types.

In any case, the main argument is not the accident of adding these methods to the public API, but their uselessness there (IMO).

mhilbrunner reviewed (approved):
PR review meeting: approved, wasn't exposed in 3.x anyway and the API is too low-level, if there is demand, we can re-add a proper one (like https://github.com/godotengine/godot/pull/67752/)

akien-mga commented:
Thanks!
